src/rotator_library/telemetry.py

```
@@ -740,6 +784,266 @@ def mark_search_key_exhausted(self, provider: str, api_key: str):
+                    reset_period,
+                    reset_date,
+                ),
+            )
+            conn.commit()
+            conn.close()
+            logger.info(
+                f"Registered LLM provider {provider} ({free_type}, "
+                f"allowance={initial_allowance})"
+            )
+
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)